src/libs/MoneyRequestUtils.ts

```diff
@@ -40,16 +47,17 @@ function addLeadingZero(amount: string): string {
 /**
  * Calculate the length of the amount with leading zeroes
  */
-function calculateAmountLength(amount: string, decimals: number): number {
+function calculateAmountLength(amount: string): number {
     const leadingZeroes = amount.match(/^0+/);
     const leadingZeroesLength = leadingZeroes?.[0]?.length ?? 0;
-    const absAmount = parseFloat((Number(stripCommaFromAmount(amount)) * 10 ** decimals).toFixed(2)).toString();
+    const absAmount = parseFloat((Number(stripCommaFromAmount(amount)) * 100).toFixed(2)).toString();
 
-    if (/\D/.test(absAmount)) {
+    // Don't allow the amount if the parsing to number results in NaN, Infinity, or with a scientific notation (e.g., 1e+26)
+    if (/NaN|Infinity|e\+/.test(absAmount)) {
         return CONST.IOU.AMOUNT_MAX_LENGTH + 1;
     }
 
-    return leadingZeroesLength + (absAmount === '0' ? 2 : absAmount.length);
+    return leadingZeroesLength + (absAmount === '0' ? 2 : stripDotFromAmount(absAmount).length);
```

Comment:
Also, another way to fix this is by passing the removed-dot-amount to the old regex

```
const amountWithoutDecimalSeparator = stripDotFromAmount(absAmount);
if (/\D/.test(amountWithoutDecimalSeparator)) {
    return CONST.IOU.AMOUNT_MAX_LENGTH + 1;
}
```

Let me know if we want to use this instead.

Reply:
I think I like the current version better, but can we also add unit tests to prevent future regressions please?

Reply:
Before adding a test, what do you think about my suggestion [here](https://github.com/Expensify/App/pull/42815#issuecomment-2138678006). Do you think we should do it or keep it as it is for now? 

If we want to do it, then we don't need the test for this specific function.